xbmc/cores/AudioEngine/Utils/AEUtil.cpp

```diff
@@ -607,3 +607,25 @@ int CAEUtil::GetAVChannelIndex(enum AEChannel aechannel, uint64_t layout)
   av_channel_layout_uninit(&ch_layout);
   return idx;
 }
+
+void CAEUtil::GenerateSilence(AEDataFormat format,
+                              unsigned int frameSize,
+                              void* buffer,
+                              unsigned int frames)
+
+{
+  const unsigned int dataLength{frames * frameSize};
+
+  switch (format)
+  {
+    case AV_SAMPLE_FMT_U8:
+    case AV_SAMPLE_FMT_U8P:
```

Comment:
Seems this should be:

```c++
    case AE_FMT_U8:
    case AE_FMT_U8P:
```

Otherwise is comparing `AEDataFormat `with `AVSampleFormat`